scripts/src/utils/rate_limit.ts

```diff
@@ -109,3 +109,18 @@ export async function assertRateLimitRegistration(app: Application) {
 	];
 	await Promise.all(limiters.map(limiter => limiter.inc(1)));
 }
+
+async function checkRateLimitUserRequests(userId: string, limit: number, duration: moment.Duration) {
+	return await checkRateLimit(`user_reqs:${ userId }:${ duration.asSeconds() }`, duration, limit, TooManyUserRequests,
+		1, 2); // use lower number of buckets to reduce stress on redis as this is called on each user request
+}
+
+export async function assertRateLimitUserRequests(user: User) {
+	const app = (await Application.get(user.appId))!;
+
+	const limiters = [
+		await checkRateLimitUserRequests(app.id, app.config.limits.hourly_user_requests || 150, moment.duration({ hours: 1 })),
+		await checkRateLimitUserRequests(app.id, app.config.limits.minute_user_requests || 20, moment.duration({ minutes: 1 }))
```

Comment:
I think you meant to pass the user ID for both calls to `checkRateLimitUserRequests`

Reply:
you are right!